client/src/pages/Dashboard/Dashboard.tsx

```diff
@@ -6,22 +6,28 @@ import { useAuth } from '../../context/useAuthContext';
 import { useSocket } from '../../context/useSocketContext';
 import { useHistory } from 'react-router-dom';
 import ChatSideBanner from '../../components/ChatSideBanner/ChatSideBanner';
-import { useEffect } from 'react';
+import { useEffect, useLayoutEffect } from 'react';
 import { Link } from 'react-router-dom';
 import Button from '@material-ui/core/Button';
 
 export default function Dashboard(): JSX.Element {
   const classes = useStyles();
 
   const { loggedInUser } = useAuth();
-  const { initSocket } = useSocket();
+  const { initSocket, disconnectSocket } = useSocket();
 
   const history = useHistory();
 
   useEffect(() => {
     initSocket();
   }, [initSocket]);
 
+  useLayoutEffect(() => {
```

Comment:
Could you elaborate the use of useLayoutEffect here?

Reply:
Ok according to the ticket. I am using useLayoutEffect  to disconnect the socket when the page is unmount. I understand I could also do it inside the useffect hook with a return statement but I was having dependency issues and since useLayoutEffect works as well I decided to stick to that.